common/notification/notification.ts

```diff
@@ -303,4 +314,57 @@ function diff(prev: any, curr: any, depth = 0) {
     return result;
 }
 
+export async function importMessageTranslations(messageTranslations: MessageTranslationFromDb[], overwrite = false) {
+    messageTranslations.sort((a, b) => a.id - b.id);
+    if (process.env.ENV !== 'dev' && overwrite) {
+        throw new Error('Cannot overwrite productive configuration');
+    }
+    let log = `Import Log ${new Date().toLocaleString('en-US')}\n`;
+
+    try {
+        await prisma.$transaction(async (prisma) => {
+            if (overwrite) {
+                const removed = await prisma.message_translation.deleteMany({});
+                log += `Through Overwrite ${removed.count} existing MessageTranslations were removed\n`;
+            }
+            const untouched = await prisma.message_translation.count({
+                where: { id: { notIn: messageTranslations.map((it) => it.id) } },
+            });
+
+            log += `${untouched} existing MessageTranslations will not be modified\n`;
+
+            for await (const translation of messageTranslations) {
```

Comment:
I never saw a `for await (...` before 😅 
What is it doing in this case?

Reply:
It's basically awaiting the loop and all the async operations in it before beginning the next loop. 
You can think of it as a `Promise.all([].map())` without the parallelization.

Reply:
Thanks for clarifying 💯